### PR TITLE
default to Java Event

### PR DESCRIPTION
**Pull request by colinsurprenant**
REPLACES #4494 which got wrongfully merged and unmerged into shared branch but now github won't believe it's not merged for real :P

- default to logstash-core-event-java
- removed dependency on Guava
- cleaned unused dependencies
- cleaned unused imports
- cleaned build.gradle
- removed java 8 code so it compiles on java 7
- use jar-dependencies to handle jackson dependency
- use json-unit for json related tests
- added gem jar creation and cleanup in build.gradle
- fix array fields nil values handling to comply with #4561
 
for merging into feature/core-event-java shared branch. @guyboertje please review.

relates to #4480

### CODE REVIEW

**Comment by colinsurprenant** [comment]
@ph @tal I merged master which includes #4561 so running 
```sh
bin/rspec logstash-core-event/spec/logstash/event_spec.rb
```
will run the `nil` values specs which the last commit before the merge commit solves

**Comment by ph** [comment]
LGTM weeee!!!!
maxresdefault

**Comment by colinsurprenant** [comment]
@ph thanks!! :D

File: logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+public class KeyNodeTest {
```
Comment (talevy) [inline logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java]: maybe add a test with `null` in the list?

**Comment by guyboertje** [comment]
@colin - Did you mean to use such an old version of Jackson?

**Comment by colinsurprenant** [comment]
@guyboertje not really, I did not really put immediate attention to Jackson because I figured we'd have to harmonize what JrJackson is using and such. In particular, if JrJackson moves to jar-dependencies, the Jackson library dependency resolution can be harmonized through that? thoughts? I mean, I can always update it to latest, wouldn't harm either.

File: logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java (hunk 1 above)
Comment (colinsurprenant) [inline logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java]: +1

Comment (colinsurprenant) [inline logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java]: and list-in-list

**Comment by guyboertje** [comment]
@colinsurprenant - No need to update, I am happy that you have a plan concerning the latest versions.

**Comment by colinsurprenant** [comment]
@guyboertje you are my plan

**Comment by guyboertje** [comment]
:smile:

File: logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java (hunk 1 above)
Comment (colinsurprenant) [inline logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java]: done

**Comment by colinsurprenant** [comment]
oh, yesterday I merged master in this branch... hence the unrelated commits.

**Comment by talevy** [comment]
LGTM

**Comment by guyboertje** [comment]
yay :shipit:

**Comment by colinsurprenant** [comment]
rebasing and merging in feature/core-event-java
